example/spec/s3Access/s3AccessSpec.js

```diff
@@ -142,53 +157,73 @@ describe('When accessing an S3 bucket directly', () => {
-      });
+    function executeTestsAgainst(testBucket) {
+      describe('while in the the same region ', () => {
+        it('the bucket contents can be listed', async () => {
+          expect(await canListObjects('us-east-1', testBucket, creds)).toBe('true');
+        });
 
-      it('a write is rejected', async () => {
-        expect(await canWriteObject('us-east-1', creds)).toBe('false');
-      });
-    });
+        it('the data can be downloaded', async () => {
+          expect(await canGetObject('us-east-1', testBucket, creds)).toBe('true');
+        });
 
-    describe('while outside the region ', () => {
-      it('the bucket contents can NOT be listed', async () => {
-        expect(await canListObjects('us-west-2', creds)).toBe('false');
+        it('a write is rejected', async () => {
+          expect(await canWriteObject('us-east-1', testBucket, creds)).toBe('false');
+        });
       });
 
-      it('the data can NOT be downloaded', async () => {
-        expect(await canGetObject('us-west-2', creds)).toBe('false');
-      });
+      describe('while outside the region ', () => {
+        it('the bucket contents can NOT be listed', async () => {
+          expect(await canListObjects('us-west-2', testBucket, creds)).toBe('false');
+        });
 
-      it('a write is rejected', async () => {
-        expect(await canWriteObject('us-east-1', creds)).toBe('false');
-      });
-    });
-  });
+        it('the data can NOT be downloaded', async () => {
+          expect(await canGetObject('us-west-2', testBucket, creds)).toBe('false');
+        });
 
-  describe('with third-party/invalid credentials', () => {
-    const thirdPartyCredentials = {
-      accessKeyId: 'AKIAIOSFODNN7EXAMPLE',
-      secretAccessKey: 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
-      sessionToken: 'FAKETOKENdfkjaf9rufjfdklajf',
-      expiration: '2019-02-26 00:08:18+00:00'
-    };
+        it('a write is rejected', async () => {
+          expect(await canWriteObject('us-east-1', testBucket, creds)).toBe('false');
+        });
+      });
+    }
 
-    it('the bucket contents can NOT be listed', async () => {
-      expect(await canListObjects('us-east-1', thirdPartyCredentials)).toBe('false');
+    describe('against protected buckets', () => {
+      executeTestsAgainst(protectedBucketName);
     });
 
-    it('the data can NOT be downloaded', async () => {
-      expect(await canGetObject('us-east-1', thirdPartyCredentials)).toBe('false');
+    // TODO [MHS, 2019-04-08] enable public test when bucket policy is applied to public buckets.
+    xdescribe('against public buckets', () => {
+      executeTestsAgainst(publicBucketName);
     });
 
-    it('a write is rejected', async () => {
-      expect(await canWriteObject('us-east-1', thirdPartyCredentials)).toBe('false');
+    describe('with third-party/invalid credentials', () => {
+      const thirdPartyCredentials = {
+        accessKeyId: 'AKIAIOSFODNN7EXAMPLE',
```

Comment:
This is a "just me" thing, but if the point here is to test invalid credentials, I'd almost rather use a value that is clearly made up via `randomId('accessKeyId')` than something that looks close to real and kinda scared me when I first saw it. Totally subjective though.

Reply:
sure.